### PR TITLE
[Messenger] Simplifying SyncTransport and fixing bug with handlers transport

### PR DESCRIPTION
**Pull request by weaverryan**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | none
| License       | MIT
| Doc PR        | not needed

This is still a WIP, because it's not quite working and tests are a TODO. However, the basic idea is there. This makes SyncTransport less "weird". It acts more like a real transport... except that it "receives" and re-dispatches its message immediately.

The bug I'm trying to fix is related to the transport-based handling config that @sroze introduced. It doesn't currently play nice with the sync transport due to the unnatural way that I made it originally.

Cheers!

### CODE REVIEW

**Comment by sroze** [comment]
The approach completely makes sense to me. When you say "it's not quite working", what's the main remaining thing to crack? 🤔

**Comment by weaverryan** [comment]
Ready for review! This indeed worked fine the whole time - I was "fooled" by an outdated cache - opened up a separate issue about that #31409

File: src/Symfony/Component/Messenger/Transport/Sync/SyncTransport.php [hunk 1]
```diff
@@ -13,18 +13,27 @@
...
 use Symfony\Component\Messenger\Exception\InvalidArgumentException;
-use Symfony\Component\Messenger\Stamp\ForceCallHandlersStamp;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\ReceivedStamp;
+use Symfony\Component\Messenger\Stamp\SentStamp;
 use Symfony\Component\Messenger\Transport\TransportInterface;
 
 /**
- * A "fake" transport that marks messages to be handled immediately.
+ * Fake transport that immediately marks messages as received and dispatches for handling.
```
Comment (OskarStark) [inline src/Symfony/Component/Messenger/Transport/Sync/SyncTransport.php]: In fact it’s not a “fake”, but just some kind of transport.
Should we remove this kind of wording?

**Review by OskarStark — APPROVED** [review]

**Comment by weaverryan** [comment]
Tests passing - this is ready to go.

**Review by Tobion — APPROVED** [review]
So clean 👍 

With that I feel we don't need the `send_and_handle` logic anymore as the sync transport does the same thing.

So the following example from the doc in https://symfony.com/doc/current/messenger.html#routing
```
framework:
    messenger:
        routing:
            'My\Message\ThatIsGoingToBeSentAndHandledLocally':
                 senders: [amqp]
                 send_and_handle: true
```
is the same as
```
framework:
    messenger:
        routing:
            'My\Message\ThatIsGoingToBeSentAndHandledLocally':
                 senders: [amqp, sync]
```
if I'm not mistaken.

**Comment by weaverryan** [comment]
I was thinking the same thing :). That would remove an awkward option.

**Comment by Tobion** [comment]
I'm looking into creating a PR.

**Comment by Tobion** [comment]
Thank you @weaverryan.

**Pull request merged**